pkg/devfile/image/docker_compatible.go

```diff
@@ -34,9 +34,21 @@ func (o *DockerCompatibleBackend) Build(image *devfile.ImageComponent, devfilePa
 	buildSpinner := log.SpinnerNoSpin("Building image locally")
 	defer buildSpinner.End(false)
 
-	shell := getShellCommand(o.name, image, devfilePath)
+	err := os.Setenv("PROJECTS_ROOT", devfilePath)
+	if err != nil {
+		return err
+	}
+
+	err = os.Setenv("PROJECT_SOURCE", devfilePath)
+	if err != nil {
+		return err
+	}
 
-	cmd := exec.Command("bash", "-c", shell)
+	shellCmd := getShellCommand(o.name, image, devfilePath)
+	for i := 0; i < len(shellCmd); i++ {
+		shellCmd[i] = os.ExpandEnv(shellCmd[i])
+	}
```

Comment:
I think below would be more readable, WDYT?:
```go
for i, cmd := range shellCmd {
	shellCmd[i] = os.ExpandEnv(cmd)
}
```

Reply:
Sure